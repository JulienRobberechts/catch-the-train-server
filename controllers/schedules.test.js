const SchedulesController = require("./schedules");
const apiAdapterTestDouble = require("../adapters/ratp-api/ratp-api-adapter.mock");

describe("schedules controller", () => {
  let controller;
  beforeAll(() => {
    controller = new SchedulesController({
      apiAdapter: apiAdapterTestDouble
    });
  });

  it("should not accept null api adapter part 1", async () => {
    const invalidControllerInit = () => new SchedulesController();
    expect(invalidControllerInit).toThrow();
  });
  it("should not accept null api adapter part 2", async () => {
    const invalidControllerInit = () => new SchedulesController({});
    expect(invalidControllerInit).toThrow();
  });
  it("should not accept null api adapter part 3", async () => {
    const invalidControllerInit = () =>
      new SchedulesController({ apiAdapter: null });
    expect(invalidControllerInit).toThrow();
  });
  it("should return schedules for a station/to", async () => {
    const now = "2020-03-10T09:22:30+01:00";
    const type = "rers";
    const line = "A";
    const station = "chatelet+les+halles";
    const result = await controller.getSchedulesForJourney(
      now,
      type,
      line,
      station
    );
    expect(result).toMatchSnapshot();
  });
});
