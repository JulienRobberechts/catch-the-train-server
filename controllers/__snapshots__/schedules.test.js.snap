// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schedules controller should return schedules for a station/to 1`] = `
Object {
  "at": "2020-03-10T09:22:30+01:00",
  "line": "A",
  "provider": "ratp",
  "routes": Array [
    Object {
      "code": "TAXE",
      "destination": "Poissy",
      "isApproaching": undefined,
      "isDeparture": undefined,
      "isOnPlatform": undefined,
      "isTerminus": undefined,
      "message": "21:46",
      "noPassenger": undefined,
      "platform": undefined,
      "time": "2020-03-10T21:46:00+01:00",
    },
    Object {
      "code": "NOTE",
      "destination": "Boissy-Saint-Leger",
      "isApproaching": undefined,
      "isDeparture": undefined,
      "isOnPlatform": undefined,
      "isTerminus": undefined,
      "message": "21:34",
      "noPassenger": undefined,
      "platform": undefined,
      "time": "2020-03-10T21:34:00+01:00",
    },
  ],
  "station": "chatelet+les+halles",
  "type": "rers",
}
`;
